tools/tdx-measure/rtmr/rtmr.go

```diff
@@ -78,8 +91,89 @@ var (
 		0xa3, 0xbc,
 		0xda, 0xd0, 0x0e, 0x67, 0x65, 0x6f,
 	}
+	fvNameGUID = [16]byte{
+		0xc9, 0xbd, 0xb8, 0x7c,
+		0xeb, 0xf8,
+		0x34, 0x4f,
+		0xaa, 0xea,
+		0x3e, 0xe4, 0xaf, 0x65, 0x16, 0xa1,
+	}
+	fileGUID = [16]byte{
+		0x21, 0xaa, 0x2c, 0x46,
+		0x14, 0x76,
+		0x03, 0x45,
+		0x83, 0x6e,
+		0x8a, 0xb6, 0xf4, 0x66, 0x23, 0x31,
+	}
 )
 
+// buildFilePathList assembles the device path passed by QEMU for boot option Boot0000.
+func buildFilePathList() ([]byte, error) {
+	var buffer bytes.Buffer
+	addNode := func(ty uint8, subType uint8, data []byte) error {
+		if err := buffer.WriteByte(ty); err != nil {
+			return err
+		}
+		if err := buffer.WriteByte(subType); err != nil {
+			return err
+		}
+		length := len(data) + 4
+		if err := binary.Write(&buffer, binary.LittleEndian, uint16(length)); err != nil {
+			return err
+		}
+		if _, err := buffer.Write(data); err != nil {
+			return err
+		}
+		return nil
+	}
+	if err := addNode(mediaProtocolType, piwgFirmwareVolumeSubType, fvNameGUID[:]); err != nil {
+		return nil, err
+	}
+	if err := addNode(mediaProtocolType, piwgFirmwareFileSubType, fileGUID[:]); err != nil {
+		return nil, err
+	}
+	if err := addNode(endOfPathType, 0xff, []byte{}); err != nil {
+		return nil, err
+	}
+	return buffer.Bytes(), nil
+}
+
+// buildEfiLoadOption assembles the EFI_LOAD_OPTION passed by QEMU for boot option Boot0000.
```

Comment:
How did you figure this out - is it documented somewhere or does one need to read the source to understand what's happening? Can we leave some references in case this changes with future qemu releases?

Reply:
I looked at the UEFI and PI specs and dissected the bytes manually. The constants are found in both QEMU and OVMF, so I don't think QEMU changing them is even possible without breaking older OVMFs.

Reply:
Thanks - do you think we could link the relevant parts?

Reply:
I added some spec links for the data structures.